patterns/disposer.md

```diff
@@ -0,0 +1,84 @@
+# The Promise Disposer Pattern
```

Comment:
@benjamingr, @petkaantonov  I would not compare it with RAII though, its too different. Its specifically designed for managing non-memory resources in a garbage collected language where RAII isn't available anymore. 

@vkurchatkin its true, we introduced this feature in bluebird and I haven't seen other libraries adopt it. I guess most people consider it trivial as its simplest implementation can be easily written with `.finally()`. 

I'd say its effective difference from `finally` is non-trivial though, because returning disposers lets API authors ensure that a resource cannot be used in any other way other than by passing it in a `using` closure. And doing that guarantees that any exceptions or errors will call the resource's disposer. Finally it also provides the one important place where node really does need to crash: if the code that releases (scarce) resources throws or asynchronously fails.